src/main/io/displayport_msp.c

```diff
@@ -168,6 +176,44 @@ static uint32_t txBytesFree(const displayPort_t *displayPort)
     return mspSerialTxBytesFree();
 }
 
+#ifdef USE_MSP_DISPLAYPORT_FONT
+static bool writeFontCharacter(displayPort_t *displayPort, uint16_t addr, const osdCharacter_t *chr)
+{
+    displayPortMspCommand_t displayPortCommand;
+
+    if (!chr) {
+        return false;
+    }
+
+    displayPortCommand.command = MSP_DP_FONTCHAR_WRITE;
+    displayPortCommand.row = addr & 0xff;
+    displayPortCommand.col = (addr >> 8) & 0xff;
+    displayPortCommand.attribute = 0;
+
+    memcpy(displayPortCommand.data, chr->data, OSD_CHAR_BYTES);
+
+    int res = output(displayPort, MSP_DISPLAYPORT, (uint8_t*)&displayPortCommand, OSD_CHAR_BYTES + 4);
+
+    // 80ms delay needed to ensure the MSP display has enough time to process the font data
+    delay(80);
```

Comment:
Rather than adding a delay it would be better to add a handshake/acknowledge.

Reply:
@SteveCEvans can we merge this and improve in a follow up maybe ?